src/components/media-importer.jsx

```diff
@@ -0,0 +1,392 @@
+	jpg: ['image/jpg'],
+	jpeg: ['image/jpeg'],
+	gif: ['image/gif'],
+	png: ['image/png'],
+	mp3: ['audio/mp3', 'audio/mpeg', 'audio/mpeg3'],
+	obj: ['application/octet-stream', 'model/obj'],
+}
+
+const SORT_OPTIONS = [
+	{
+		sortMethod: sortString.bind(null, 'name'), // bind the field name to the sort method
+		name: 'Title',
+		field: 'name',
+	},
+	{
+		sortMethod: sortString.bind(null, 'type'), // bind the field name to the sort method
+		name: 'Type',
+		field: 'type',
+	},
+	{
+		sortMethod: sortNumber.bind(null, 'timestamp'), // bind the field name to the sort method
+		name: 'Date',
+		field: 'timestamp',
+	},
+]
+
+const MediaImporter = () => {
+	const queryClient = useQueryClient()
+	const [selectedAsset, setSelectedAsset] = useState(null)
+	const [sortAssets, setSortAssets] = useState(null) // Display assets list
+	const [sortState, setSortState] = useState({
+		sortAsc: false, // Sorted list in asc or desc
+		sortOrder: 0, // List sorting options
+	})
+	const [assetList, setAssetList] = useState({})
+	const [showDeletedAssets, setShowDeletedAssets] = useState(false)
+	const [filterSearch, setFilterSearch] = useState('') // Search bar filter
+
+	const { data: listOfAssets } = useQuery({
+		queryKey: ['media-assets', selectedAsset],
+		queryFn: () => apiGetAssets(),
+		staleTime: Infinity,
+		onSettled: (data) => {
+			if (!data) console.warn('Error in asset retrieval')
+			else {
+				const list = data.map(asset => {
+	
+					const creationDate = new Date(asset.created_at * 1000)
+					return {
+						id: asset.id,
+						type: asset.type,
+						name: asset.title.split('.').shift(),
+						timestamp: asset.created_at,
+						thumb: _thumbnailUrl(asset.id, asset.type),
+						created: [creationDate.getMonth(), creationDate.getDate(), creationDate.getFullYear()].join('/'),
+						is_deleted: parseInt(asset.is_deleted)
+					}
+				})
+
+				list.forEach((asset) => {
+					if (asset.id == selectedAsset) _loadPickedAsset(asset)
+				})
+
+				setAssetList(list)
+			}
+		}
+	})
+
+	/****** hooks ******/
+
+	// useEffect(() => {
+	// 	console.log(selectedAsset)
+	// 	if (selectedAsset && assetList.length) {
+	// 		console.log('selected asset check')
+	// 		assetList.forEach((asset) => {
+	// 			console.log(asset)
+	// 			if (asset.id == selectedAsset) _loadPickedAsset(asset)
+	// 		})
+	// 	}
+	// }, [selectedAsset])
+
+	// Asset list, sorting, search filter, or show delete flag is updated
+	// Processes the list sequentially based on the state of each
+	useEffect(() => {
+		if (!assetList || !assetList.length) return
+		
+		// first pass: filter out deleted assets, if we're not displaying them
+		let listStageOne = assetList.filter((asset) => (!showDeletedAssets && !asset.is_deleted) || showDeletedAssets )
+
+		// second pass: filter assets based on search string, if present
+		let listStageTwo = filterSearch.length ? listStageOne.filter((asset) => asset.name.toLowerCase().match(filterSearch)) : listStageOne
```

Comment:
filterSearch is not being converted to lower case in addition to asset.name (searching in uppercase will not return any results)

Reply:
Good catch, resolved :+1: